python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,248 @@
+            return
+
+        for gt_index, gt in enumerate(self._true_y):
+            for detect_index, detect in enumerate(sorted_list):
+                iou_score = torchvision.ops.box_iou(
+                    Tensor(detect[1:5]).unsqueeze(0).view(-1, 4),
+                    Tensor(gt[1:5]).unsqueeze(0).view(-1, 4))
+
+                if iou_score.item() == 0:
+                    # if iou is 0, then prediction is detecting the background
+                    self._match_matrix[gt_index][detect_index] = (
+                        ErrorLabelType.BACKGROUND)
+                    continue
+                elif self._iou_threshold <= iou_score:
+                    # the detection and ground truth bb's are overlapping
+                    if detect[0] != gt[0]:
+                        # the bboxes line up, but labels do not
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.CLASS_NAME)
+                        continue
+                    elif (ErrorLabelType.MATCH in
+                          self._match_matrix[gt_index]):
+                        # class name and bbox correct, but there is already a
+                        # match with a higher confidence score (this is why
+                        # it was imporant to sort by descending confidence
+                        # scores as the first step)
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.DUPLICATE_DETECTION)
+                        continue
+                    else:
+                        # this means bboxes overlap, class names = (1st time)
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.MATCH)
+                        continue
+                else:
+                    if detect[0] != gt[0]:
+                        # the bboxes don't line up, and labels do not
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.CLASS_LOCALIZATION)
+                        continue
+                    else:
+                        # the bboxes don't line up, but the labels are correct
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.LOCALIZATION)
+                        continue
+
+            # resort the columns (so no longer ordered by descending conf
+            # scores)
+            self._match_matrix[gt_index] = [self._match_matrix[gt_index][i]
+                                            for i in original_indices]
+
+    def compute_error_list(self):
+        """
+        Determine a complete list of errors encountered during prediction.
+
+        Note that it is possible to have more errors than actual objects
+        in an image (because we account for missing detections and
+        duplicate detections).
+        """
+        self.compute()
```

Comment:
why not just remove the match_matrix from the class and return it here from the method?  it seems like it doesn't really need to be cached anyway, since just the error list itself matters.  I think this is because you originally wrote this as a manager and then removed that, but the code still follows the style of the managers which is unnecessary here.